impress-org/givewp
# Feature: add honorific property to donation model

Resolves [GIVE-116]

## Description



Following up from recent PR #7101, this adds a new `honorific` property to the donation model in order to preserve the field value on the donation record.  This is a new concept for GiveWP, as previously the honorific field always updated the donor record.  This means, every donation will store the honorific field (if present) - and only _new_ donors will include this field in its own meta.

In a subsequent PR, we can choose to actually implement this new property.

## Affects



- The donation model
- During v2 & v3 forms - the honorific field will now be persisted in donation meta `_give_donor_billing_title_prefix`

## Visuals



<img width="700" alt="Screenshot 2023-11-27 at 2 46 36 PM" src="https://github.com/impress-org/givewp/assets/10138447/4089880f-e40a-4fd6-b583-dca9688abce7">

## Testing Instructions



- Make a donation - filling out the honorific field, using both v2 and v3 forms
- Check the database to make sure `_give_donor_billing_title_prefix` exists and accurate

## Pre-review Checklist






[GIVE-116]: https://stellarwp.atlassian.net/browse/GIVE-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
